### ISSUE TITLE
Impossible to migrate from local Realm to Realm Cloud

### ISSUE THREAD

**Issue by selyon** [realm/realm-java#6639]
#### Goal
I have an application that runs with Realm, data is stored locally and it work fine.
Now I want to add the cloud feature but it's not possible.

#### Actual Results
When I install the new app over the old one, I get this error
E/REALM_JNI: jni: ThrowingException 1, Provided schema version 0 is less than last set version 5
The problem here is that with Realm Cloud there is no way to set the schema versione, SyncConfiguration doesn't have a method to do it.
The only way of making this work is to uninstall the old app, but I need to keep the application data so I can't do it.

#### Version of Realm and tooling

Realm version(s): 5.15.2

Realm Sync feature enabled: Yes/No

Android Studio version: 3.5.1

Android Build Tools version: 3.1.4

Gradle version: 4.4

Which Android version and device(s): 4.4

**Comment by cmelchior** [comment]
Unfortunately, there currently doesn't exist a way to automatically upgrade a local Realm to a synchronized Realm with existing data. You need to manually copy all data over. This might or might not be relatively easy depending on your data model.

**Comment by selyon** [comment]
What do you mean with "copy all data"? There a way to transfer data from my local Realm to the copy on cloud before installing the new application?

**Comment by cmelchior** [comment]
No, you would need to do it the first time people log into the app after they upgrade

**Comment by Zhuinden** [comment]
The safest (although not necessarily sanest) way to do it would be to open the Sync Realm using the typed api, and copy the data over from the old Realm if it exists using the DynamicRealm API.

That way, even if you make changes to your schema, you won't need to mess with migration of your non-sync Realm.

**Comment by cmelchior** [comment]
Converting this to a feature request for automatically upgrading a local Realm to a synchronized Realm. We should be able to relatively easy make an API for this at the core level that will copy everything from a local Realm to a synchronized Realm thus creating the correct history.
